### PR TITLE
[async-lock] Update types and tests

### PR DESCRIPTION
**Pull request by Rhymmor**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/rogierschouten/async-lock/blob/master/lib/index.js
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/async-lock/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-// Type definitions for async-lock
+// Type definitions for async-lock 1.1.0
```
Comment (DanielRosenwasser) [inline types/async-lock/index.d.ts]: Looks like our lint rules are disabled for this package. Can you enable the lints in our `tslint.json` and fix up the version string to only be a `MAJOR.MINOR` version (e.g. `4.7`), as opposed to including the patch or containing a leading `v` (e.g. `v1.2.3`).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
Can you also fix up the modified file permissions in each of these files (move back from `755` to `644`)?

https://stackoverflow.com/questions/1257592/how-do-i-remove-files-saying-old-mode-100755-new-mode-100644-from-unstaged-cha

(As a tip: this has been a problem when I've written files over Samba/SMB so that's possibly the cause. I set my `map archive`, `map system`, and `map hidden` to `no` explicitly. See http://www.oreilly.com/openbook/samba/book/ch05_03.html for more details).

File: types/async-lock/index.d.ts (hunk 1 above)
Comment (Rhymmor) [reply types/async-lock/index.d.ts]: ok. I've enabled all linting rules: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23340/commits/2623588675a7b5ea2c3845edb8af7df9b13840c0

**Comment by Rhymmor** [comment]
@DanielRosenwasser Sure, fixed permissions: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23340/commits/d52f4f7b327af576cd33174d48602ad4ce47f9aa

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thank you for your patience!

**Pull request merged**